src/ctap/mod.rs

```diff
@@ -146,6 +150,52 @@ fn truncate_to_char_boundary(s: &str, mut max: usize) -> &str {
     }
 }
 
+/// Parses the metadata of an upgrade, and checks its correctness.
+///
+/// Returns the hash over the upgrade, including partition and some metadata.
+/// The metadata layout is:
+/// | -- 32B upgrade hash (SHA256) -- | 8B other |
+fn parse_metadata(
+    upgrade_locations: &UpgradeLocations,
+    metadata: &[u8],
+) -> Result<[u8; 32], Ctap2StatusCode> {
+    const METADATA_LEN: usize = 40;
+    if metadata.len() < METADATA_LEN {
```

Comment:
"<" or "!="?
Is the metadata field valid if it's 50 bytes?

Reply:
The logic as it is now leaves some freedom to either call with a whole page, or only call with a prefix. The page will than implicitly be overwritten with a 0xFF padding. We could change that to only accept exact slices. Maybe a tighter interface with less room for clients makes it more maintainable?

Reply:
I liked the idea of a more precise input, and implemented it. PTAL